src/Xamarin.Android.Tools.Bytecode/ClassPath.cs

```diff
@@ -251,13 +251,15 @@ JavaDocletType GetDocletType (string path)
 				kind = JavaDocletType.Java8;
 
 			// Check to see if it's an api.xml formatted doc
-			string rawXML = null;
-			using (var reader = File.OpenText (path)) {
-				int len = reader.ReadBlock (buf, 0, buf.Length);
-				rawXML = new string (buf, 0, len);
+			if (File.Exists (path)) {
```

Comment:
So, the `ClassParse` [task just passes](https://github.com/Redth/xamarin-android/blob/master/src/Xamarin.Android.Build.Tasks/Tasks/ClassParse.cs#L60) in `DocumentationPaths` strings which usually is a directory, but [_could_ be a file](https://github.com/xamarin/xamarin-android/pull/920/files#diff-17dd9a4d2f58f386c791efbeec36241cR34), as the `ApiXmlDocPaths` is intended to be a path to an xml *file*.

I think this fix should work in this case since `File.Exists` will indeed skip over directories and this code is only hit for files.

Reply:
Is there a quick test we could add with a file? Looking at the code, I think this part would fail if a file path came in:

```
using (var reader = File.OpenText (Path.Combine (path, "index.html")))
    reader.ReadBlock (buf, 0, buf.Length);
```